src/Compilers/CSharp/Portable/BoundTree/BoundImplicitIndexerAccess.cs

```diff
@@ -29,5 +29,11 @@ internal BoundExpression GetReceiver()
             Debug.Assert(receiver is not null);
             return receiver;
         }
+
+        private partial void Validate()
+        {
+            Debug.Assert(LengthOrCountAccess is BoundPropertyAccess or BoundLocal);
```

Comment:
>BoundPropertyAccess or BoundLocal [](http://example.com/codeflow?start=48&length=33)

It looks like CheckValue can return a BoundBadExpression. We might want to be prepared to deal with that.